File: result/translate/chatgpt/python2java/java/BINARY_SEARCH.java
public static int f_gold(int[] arr, int l, int r, int x) {
    if (r >= l) {
        int mid = l + (r - l) / 2;
        if (arr[mid] == x) {
            return mid;
        } else if (arr[mid] > x) {
            return f_gold(arr, l, mid - 1, x);
        } else {
            return f_gold(arr, mid + 1, r, x);
        }
    } else {
        return -1;
    }
}

File: result/translate/chatgpt/java2python/python/BINARY_SEARCH.py
def f_gold(arr, l, r, x):
    if r >= l:
        mid = l + (r - l) // 2
        if arr[mid] == x:
            return mid
        if arr[mid] > x:
            return f_gold(arr, l, mid - 1, x)
        return f_gold(arr, mid + 1, r, x)
    return -1